superset/connectors/sqla/views.py

```diff
@@ -369,6 +371,15 @@ class RowLevelSecurityFiltersModelView(  # pylint: disable=too-many-ancestors
         add_form_query_rel_fields = app.config["RLS_FORM_QUERY_REL_FIELDS"]
         edit_form_query_rel_fields = add_form_query_rel_fields
 
+    @staticmethod
+    def is_enabled() -> bool:
+        return is_feature_enabled("ROW_LEVEL_SECURITY")
+
+    @before_request
+    def ensure_enabled(self) -> None:
+        if not self.is_enabled():
+            raise NotFound()
```

Comment:
We could potentially generalize this by breaking this out into a Mixin that checks `is_feature_enabled` on a reserved property. Something like this:

```python
class FeatureFlagMixin:
    feature_flag: str

    def is_enabled(self) -> bool:
        return is_feature_enabled(self.feature_flag)

    @before_request
    def ensure_enabled(self) -> None:
        if not self.is_enabled():
            raise NotFound()
```

Then later this could be used as follows:

```python
class RowLevelSecurityFiltersModelView(  # pylint: disable=too-many-ancestors
    FeatureFlagMixin, SupersetModelView, DeleteMixin
):
    feature_flag = "ROW_LEVEL_SECURITY"
    ....
```
we could potentially also add support for callbacks that make it possible to add custom validators that don't rely on `is_feature_enabled`:

```python
class FeatureFlagMixin:
    feature_flag: Union[str, Callable[[], str]]

    def is_enabled(self) -> bool:
        if callable(self.feature_flag):
            return self.feature_flag()
        return is_feature_enabled(self.feature_flag)

    @before_request
    def ensure_enabled(self) -> None:
        if not self.is_enabled():
            raise NotFound()
```

and then use it like this:

```python
    feature_flag = lambda _: app.config["FAB_ADD_SECURITY_VIEWS"] and app.config["SUPERSET_LOG_VIEW"]
```

Reply:
I think we could, but it's a little bit more complicated then that. `@before_request` needs to be able to be scoped to a subset of route handlers in some cases. We can achieve that, but for now I think we should ship the more naive version. If this pattern continues to be used moving forward, the opportunity for a refactor should present itself.

Reply:
Gotcha - I don't have the full picture of the context in which this change is used, so some of my comments will probably be irrelevant 👍